esmvaltool/diag_scripts/seaice_area_extents/seaice_mapextents.py

```diff
@@ -0,0 +1,161 @@
+    latmax = obs_si.lat.max().values.item()
+
+    proj = SouthPolarStereo(true_scale_latitude=-70)
+
+    # fig set up, width for 2 models, check len mod_si_ls
+    figure = plt.figure(figsize=(9, len(months) * 4))
+    j = 0  # to iterate through positions on figure
+
+    for mon in months:
+        cdr = obs_si.siconc.sel(
+            time=obs_si.siconc.time.dt.month.isin(mon)).mean('time')
+        i = 1
+        for mod_label, mod_si in mod_si_ls.items():
+
+            diff_ds, mod_regrid = model_regrid_diff(mod_si, obs_si, cdr, mon,
+                                                    latmax)
+            # save plot data
+            save_data(mod_label + calendar.month_abbr[mon] + '_mean', prov,
+                      cfg, mod_regrid.to_iris())
+            save_data(mod_label + calendar.month_abbr[mon] + '_obs_diff', prov,
+                      cfg, diff_ds.to_iris())
+
+            axes = plt.subplot(len(months), 3, i + j * 3, projection=proj)
+
+            diffmap = axes.contourf(diff_ds.x,
+                                    diff_ds.y,
+                                    diff_ds,
+                                    levels=np.arange(-90, 91, 20),
+                                    cmap='RdBu')
+            cs_cdr = cdr.plot.contour(levels=[15], ax=axes)
+            cs_mod = mod_regrid.plot.contour(levels=[15],
+                                             ax=axes,
+                                             colors=['black'])
+
+            plt.title(calendar.month_abbr[mon] + ' ' + mod_label)
+
+            i += 1
+        j += 1
+
+    line_cdr = mlines.Line2D([], [],
+                             color=cs_cdr.collections[0].get_edgecolor(),
+                             label="Observed Extent")
+
+    line_mod = mlines.Line2D([], [],
+                             color=cs_mod.collections[0].get_edgecolor(),
+                             label="Modelled Extent")
+
+    plt.legend(handles=[line_cdr, line_mod],
+               loc='center left',
+               bbox_to_anchor=(1.2, 0.5))
+    cax = plt.axes([0.7, 0.55, 0.04, 0.3])
+    _ = plt.colorbar(diffmap,
+                     cax=cax,
+                     label='Difference in \nSea Ice Concentration')
+    return figure
+
+
+def main(cfg):
+    """Compute."""
+    # Get a description of the preprocessed data that we will use as input.
+    input_data = cfg['input_data'].values()
+    data = []
+
+    # Find input datasets to use
+    for dataset in input_data:
+
+        input_file = [dataset['filename'], dataset['dataset']]
+        # drop areacello dataset for map
+        if dataset['short_name'] == 'siconc':
+            data.append(input_file)
+
+    inputs_df = pd.DataFrame(data, columns=['filename', 'dataset'])
+
+    logger.info(inputs_df)
+    mod_si_dict = {}
+    for filepath, data_name in inputs_df.itertuples(index=False):
+        if data_name == 'NSIDC-G02202-sh':
+            # Load the data
+            obs_si = xr.open_dataset(filepath)
```

Comment:
they would be netcdf files. There's a CMORizer for the dataset `NSIDC-G02202-sh`. Mainly using xarray here as porting code from a "COSIMA" notebook instead of trying to rewrite something... it could be rewritten to use the esmvaltool framework better though 🤔

Reply:
I'd argue that'd be nice - either import esmvalcore and use its loading function, or, if the data is not CMOR compliant (should be, if it was cmorized in advance), but it is CF-compliant, just use `iris` :+1:

Reply:
well we needed to load as xarray dataset as all the code for data analysis, calculations, regridding is done with xarray syntax so that would all have to be rewritten to use iris. though if we edit to use a different regridder that would have to be done too